### DISCUSSION TITLE
How to select requested-only fields from DB with Ariadne GraphQL ?

### DISCUSSION THREAD

**Discussion by kleysonr** [mirumee/ariadne#1169] [Q&A] [answered]
How to request only the required fields from DB in resolvers.

Let's say there's a request of a basic user info:

```
{
  users {
    id
    name
  }
}
```
And the DB structure for user entity is much more complex, wide and "data-heavy". Now one would obviously trying to avoid getting all of the user info from a DB just to filter it on a server side and pass to a client only ids and names requested.

I was wondering what solutions are considered to be optimal in this case (or is it even possible to make a universal solution not bound to a specific data structure, as there might be some additional data loads and joining involved).

Yes, it could be done manually by getting requested fields from the resolver's info argument.

Btw I'm also currently not bound to a specific DB or a data-provider like SQLAlchemy, as I'm trying to wrap my head around the abstract use-case. 

I would really appreciate your thoughts and experience on this one. Either abstract or with actual providers and databases.

Thanks.

**Comment by rafalp** [comment, marked as answer]
You have two options here:

- look up fields in `info` arg
- look up whole query contents in custom `root_value` callable (this is what Ariadne GraphQL Proxy does)

I vaguely remember a library existing in past for Graphene and Django that would build querysets based on query contents, but I also remember it was found very fast to be slow to work with.

**Comment by pylipp** [comment]
This problem is also discussed in #1116
